src/prototype_creep_routing.js

```diff
@@ -267,6 +267,10 @@ Creep.prototype.moveByPathMy = function(route, routePos, start, target, skipPreM
         if (this.memory.killPrevious) {
           this.killPrevious();
         }
+        if (this.memory.checkRecycle && this.isStuck() && !this.memory.routing.reverse) {
```

Comment:
I do not like it a lot. carry may stuck not only because of lack of resources at sourcer
either way, name `recycleOnStuck` would be better then `checkRecycle`

Reply:
maybe we may move it to carries pre-move?

Reply:
It's a little bit tricky to move this to pre-move because I need to use `path.length`.  Maybe we can store `path.length` in creeps' memory.

Reply:
we can pass it the same way as directions